### DISCUSSION TITLE
After calling OpcUaServer.shutdown(), I have leaking threads named: milo-netty-event-loop-x

### DISCUSSION THREAD

**Discussion by fatih-algan** [eclipse-milo/milo#1177] [General]
Hi,

After I terminate the OpcUaServer via calling shutdown(), I see that the shared event loop threads created by the NioEventLoopGroup are not terminated. The event loop group is created by the sharedEventLoop() method of the org.eclipse.milo.opcua.stack.core.Stack class.

Is this a bug or what do I need to do to have those threads get cleaned up/terminated properly?

Thanks in advance.

**Comment by kevinherron** [comment]
When your application is done with Milo entirely you can call one of the `Stack.releaseSharedResources` functions.

https://github.com/eclipse/milo/blob/f1ce134e2ffbf15f4c03269d62235985aa34fbc4/opc-ua-stack/stack-core/src/main/java/org/eclipse/milo/opcua/stack/core/Stack.java#L192-L197
